src/common/utils.test.ts

```diff
@@ -50,5 +50,12 @@ describe("Helper functions", () => {
 
             expect(decryptedProject).toEqual(testProject);
         });
+
+        it("Replaces an extension with new extension", () => {
+            expect(replaceExtension("filename.badext", ".jpg")).toEqual("filename.jpg");
+            expect(replaceExtension("filename.this.is.bad", ".jpg")).toEqual("filename.this.is.jpg");
+            expect(replaceExtension("filename", ".jpg")).toEqual("filename");
```

Comment:
if a file has no extension, should we add an extension?

Reply:
Moved fix into `HtmlFileReader` and added tests.  Will always add "jpg" extension if it does not already exist.